### PR TITLE
Feature/proj 63

### PR DESCRIPTION
**Pull request by Emilo067**
#### Overview
This PR introduces a new `ConfirmEmail` component to the project. The component is responsible for displaying a confirmation message to the user once their email has been successfully verified. It features a responsive design to ensure a seamless user experience across a variety of devices, from desktop monitors to mobile phones.

#### Component Features
- **Typography**: Utilizes the `Typography` component from `@photo-fiesta/ui-lib` to render the title and confirmation message. The title "Congratulations!" is styled prominently to catch the user’s attention, followed by a confirmation message in a smaller text size.
- **Button**: Includes a `Button` component that allows the user to proceed by signing in after their email has been confirmed. The button is centrally aligned and adapts its size based on the screen width to maintain usability on smaller screens.
- **Image**: An illustration (`bro`) from the assets directory is included using the `Image` component from Next.js. This image is meant to visually reinforce the success of the email confirmation and scales dynamically to fit different screen sizes.

#### Layout and Responsiveness
- **Root Container**:
  - The entire component is wrapped in a root container that is centrally aligned both vertically and horizontally within the viewport. 
  - The container's width is capped at 432px to prevent it from becoming too wide on large screens, and it adjusts to a smaller maximum width of 330px on screens with a width of 360px or less.
- **Main Section**:
  - The `main` section contains both the button and the image, arranged in a column layout by default. The elements are spaced apart by 72px on larger screens.
  - On smaller screens (360px or less), the layout changes to a column-reverse, placing the button below the image to maintain an intuitive flow.
  - The image size is also adjusted on smaller screens to fit within the reduced width, ensuring it remains visible without requiring horizontal scrolling.
- **Description Section**:
  - This section includes the title and message, centered with a gap of 20px between them. The gap is reduced to 13px on smaller screens, and the section's bottom margin is increased to 66px for better spacing.
- **Button**:
  - The button is designed to fit within the screen's width, with a maximum width of 180px on larger screens. On smaller screens, the button expands to match the width of the image, ensuring consistency and ease of use.

#### Styling Specifics
- **Flexible Layout**:
  - The `flex-direction` of the main content switches between `column` and `column-reverse` depending on the screen size, ensuring the most logical and user-friendly arrangement of elements.
  - Flexbox is heavily used to align and justify elements, providing a robust structure that adapts to various screen sizes and orientations.

#### Media Queries
- **Max-Width: 360px**:
  - At this breakpoint, the layout adapts to fit narrower screens by switching the order of elements and adjusting sizes. The button is moved below the image to align with common mobile UI patterns.
  - Both the image and button are resized to fit within the smaller screen space, with the image shrinking to 330px in width and 230px in height, and the button stretching to match the new image width.

#### Screenshots
Include screenshots below to illustrate how the `ConfirmEmail` component appears on different screen sizes:

- **Desktop View (>= 768px)**:
  - Desktop View Screenshot

- **Mobile View (<= 360px)**:
  - Mobile View Screenshot

### CODE REVIEW

File: src/assets/index.ts [hunk 1]
```diff
@@ -1 +1,2 @@
 export * from './icons'
```
Comment (Lelik7777) [inline src/assets/index.ts]: предлагаю папку assets remove to src/shared
вопрос обсуждаем

если согласен, то сразу же перекидывай

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss [hunk 2]
```diff
@@ -0,0 +1,62 @@
+.root {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+
+  width: 100%;
+  max-width: 432px;
+  height: 100vh;
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: ты с высотой точно проверил, ведь there header with height 60px?

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      <div className={classNames.description}>
+        <Typography as={'h1'} className={classNames.title} variant={'h1'}>
+          Congratulations!
+        </Typography>
+        <Typography as={'p'} variant={'text16'}>
+          Your email has been confirmed
+        </Typography>
+      </div>
+      <div className={classNames.main}>
+        <Button className={classNames.button} onClick={onClickHandler}>
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.tsx:49]: здесь кнопка  asChild=true
внутри <Link><Typography>текст

поэтому onClick не нужен и  onClickHandler too

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.main {
+  display: flex;
+  flex-direction: column;
+  gap: 72px;
+  align-items: center;
+  justify-content: center;
+
+  .img {
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.module.scss:45]: поменяй стили для картинки плиз на резиновый вариант

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+.description {
+  display: flex;
+  flex-direction: column;
+  gap: 20px;
+  align-items: center;
+
+  margin-bottom: 54px;
+
+  @media (width <= 360px) {
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: in _mixins create mixin for media

@mixin media($width) {
  @media (max-width: $width) {
    @content;
  }
}

и здесь уже используй миксин

create global variable mobile-width

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+          Sign In
+        </Button>
+        <Image alt={'confirm'} className={classNames.img} src={bro} />
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.tsx:52]: add attribute  layout="responsive"

**Review by EvgeniyaYakunina — APPROVED** [review]

File: src/assets/index.ts (hunk 1 above)
Comment (Lelik7777) [inline src/assets/index.ts]: export * from './icons' 
а почему экспорт из icons?

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import { bro } from '@/assets'
+import { Button, Typography } from '@photo-fiesta/ui-lib'
+import Image from 'next/image'
+
+import styles from './confirmEmail.module.scss'
+
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.tsx:6]: add js doc comments with example of using this component

File: src/assets/index.ts (hunk 1 above)
Comment (Emilo067) [reply src/assets/index.ts:1]: перекинул

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss (hunk 2 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx (hunk 3 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.tsx:49]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss (hunk 4 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.module.scss:45]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss (hunk 5 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx (hunk 6 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.tsx:52]: когда ставлю его изображение не соответствует размерам в стилях и слишком мелким становится

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss [hunk 8]
```diff
@@ -6,10 +9,10 @@
 
   width: 100%;
   max-width: 432px;
-  height: 100vh;
+  height: calc(100vh - 60px);
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: Эмиль, забыл сразу сказать ,создай плиз на 60px global variable header-height

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss [hunk 9]
```diff
@@ -36,27 +43,32 @@
...
 
-  @media (width <= 360px) {
+  @include media($mobile-width) {
     flex-direction: column-reverse;
     gap: 46px;
 
     .img {
-      width: 330px;
+      max-width: 330px;
       height: 230px;
```
Comment (Lelik7777) [inline src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: по высоте, ты не пробывал относительные еденицы, чтобы не харкодить высоту. Это может ломать макет в некоторых ситуациях. Это касается всех картинок

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx (hunk 6 above)
Comment (Lelik7777) [reply src/features/auth/ui/confirmEmail/confirmEmail.tsx:52]: ок

File: src/assets/index.ts (hunk 1 above)
Comment (Emilo067) [reply src/assets/index.ts]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.tsx (hunk 7 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.tsx:6]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss (hunk 8 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: исправил

File: src/features/auth/ui/confirmEmail/confirmEmail.module.scss (hunk 9 above)
Comment (Emilo067) [reply src/features/auth/ui/confirmEmail/confirmEmail.module.scss]: исправил

**Pull request merged**